### PR TITLE
fix: fix copy/delete enb/game files buttons

### LINKED ISSUE #37: Configure ENB view
# Configure enb view

Add a view to configure available enbs.

- [ ] New enb profile - Creates a new directory to store enb configuration in
- [ ] Load enb profile
- [ ] Open enb
- [ ] Delete enb

### Additional features
- [ ] Detect if path to an already defined enb is invalid on app startup?
- [ ] Provide default location for storing enbs?
- [ ] Prevent the user from selecting their own directory (and just use one internal to the app) to prevent users from accidentally renaming paths to that enb?

### Outstanding questions

- ~What does load enb do? How is this different to new?~
- ~What does delete enb do? Does this actually delete the enb or just remove it from the launcher?~
Anwered below

### LINKED ISSUE #53: ENB menu action
Add a menu item to select from a list of enb profiles.
These profiles will have been previously defined in #37 

### Outstanding questions
- What does this setting actually change? Are some files moved or is the intention for this to only be applied when the game is running?
- If it's only applied when the game is running, does this mean users can only use the launcher to launch UltSky or is the intention to continue to be able to use MO2 natively?

### LINKED ISSUE #192: Copy game folder files when the install completes
Once the Wabbajack installation has been completed, automatically copy the game folder files into the Skyrim directory.

### LINKED ISSUE #215: Add ENB Dropdown to allow users to customize ENB
Options would be Stock ENB, Custom ENB, No ENB
Stock ENB would check if ENB files are present on launch, and if not, copy them over
Custom ENB wouldn't touch ENB files at all (if a user selects this it would be assumed they know what they're doing)
No ENB would scan for any ENB files present on launch and delete them if they are

### LINKED ISSUE #290: Game files / enb files don't copy correctly
The current launcher is expecting the LE game files/enb files rather than the new SE ones.

Currently, all files are in the same directory and the ENB files will be unknown to the launcher.

The new version will use the "Stock Game" feature from Wabbajack.
The enb files will be in `MO2/ENB Presets` and all game files will already be included in the Stock Game.

- Remove the game files button and functionallity
- Update enbs to be read from `MO2/ENB Prsets`
- Remove need to input Skyrim directory

### PR DESCRIPTION
**Pull request by MattLish**
- Remove setting Skyrim directory
- Read ENB files from ENB Presets
- Make the launcher agnostic of enb files

closes #290 
closes #215
closes #192 
closes #53 
closes #37

### CODE REVIEW

**Comment by MattLish** [comment]
This needs testing properly before it can be merged

**Comment by thomasblasquez** [comment]
I think that `Skyrim.ts` and `files.ts` could be combined into a variable utility file for values we calculate once but use multiple times, rather than giving each their own file. Could even be exported from `config.ts` to avoid creating a new file. When I get home, I'll test out moving in/deleting the ENB, as well as using a custom ENB.

**Comment by MattLish** [comment]
> I think that `Skyrim.ts` and `files.ts` could be combined into a variable utility file for values we calculate once but use multiple times, rather than giving each their own file. Could even be exported from `config.ts` to avoid creating a new file. When I get home, I'll test out moving in/deleting the ENB, as well as using a custom ENB.

@RingComics 

Personally, I try to avoid "utility" files as they just become dumping grounds for all kinds of unrelated methods.
I could be persuaded that the Skyrim stuff could go into the config and `files.ts` be renamed to something more appropriate like `fileUtils.ts`. 
I originally created the `Skyrim.ts` as I foresaw more things ending up in there. Though perhaps not.

File: src/main/modOrganizer.ts [hunk 1]
```diff
@@ -2,39 +2,38 @@ import path from "path";
...
+export const launchMO2 = () => {
+  try {
+    logger.info("Launching MO2");
+    const moPath = path.join(
       userPreferences.get(USER_PREFERENCE_KEYS.MOD_DIRECTORY),
-      userPreferences.get(USER_PREFERENCE_KEYS.SKYRIM_DIRECTORY)
+      MO2EXE
     );
+    logger.debug(`MO2 path: ${moPath}`);
+    childProcess.exec(`"${moPath}"`);
```
Comment (thomasblasquez) [inline src/main/modOrganizer.ts:28]: changing this line should allow the correct profile to be opened, if not we can edit the ini above this.
```ts
childProcess.exec(`"${moPath} -p '${userPreferences.get(USER_PREFERENCE_KEYS.PRESET)}'"`);
```

Comment (MattLish) [reply src/main/modOrganizer.ts:28]: I am fairly sure that doesn't work when opening MO2. I think the profile option is only for launching the game.

I will try test it later if I can

Comment (thomasblasquez) [reply src/main/modOrganizer.ts:28]: I'm only about 50% sure it'll work, but it's worth a test. If it doesn't work, I'll write some code to edit the ini in a new PR, as I've already done similar code in Azura's Star.

**Comment by thomasblasquez** [comment]
> Personally, I try to avoid "utility" files as they just become dumping grounds for all kinds of unrelated methods. I could be persuaded that the Skyrim stuff could go into the config and `files.ts` be renamed to something more appropriate like `fileUtils.ts`. I originally created the `Skyrim.ts` as I foresaw more things ending up in there. Though perhaps not.

I don't foresee us needing more functions specifically for Skyrim, the launcher needs very little interaction with the game itself outside of ENB file manipulation. Preferably we should use MO2 for any interaction with Skyrim if we need it, like if we need to make edits to Skyrim's configuration `.ini`s. 

What additional functionality would `files.ts`/`fileUtils.ts` have going forward? If it's just the DSStore method, it might be better to move it into another module. 1-line scripts feel like a (admittedly tiny) waste of space. At this point I'm nitpicking though, it isn't that big of a deal, the code is solid and easily readable, at least for me. We should start thinking about having better documentation if we want to pivot towards using NPM modules, however.

**Comment by thomasblasquez** [comment]
This PR would also close #291

**Comment by MattLish** [comment]
> > Personally, I try to avoid "utility" files as they just become dumping grounds for all kinds of unrelated methods. I could be persuaded that the Skyrim stuff could go into the config and `files.ts` be renamed to something more appropriate like `fileUtils.ts`. I originally created the `Skyrim.ts` as I foresaw more things ending up in there. Though perhaps not.
> 
> I don't foresee us needing more functions specifically for Skyrim, the launcher needs very little interaction with the game itself outside of ENB file manipulation. Preferably we should use MO2 for any interaction with Skyrim if we need it, like if we need to make edits to Skyrim's configuration `.ini`s.

I think you're right actually, I'll move this into the config.

> What additional functionality would `files.ts`/`fileUtils.ts` have going forward? If it's just the DSStore method, it might be better to move it into another module. 1-line scripts feel like a (admittedly tiny) waste of space. At this point I'm nitpicking though, it isn't that big of a deal, the code is solid and easily readable, at least for me. We should start thinking about having better documentation if we want to pivot towards using NPM modules, however.

I am pretty sure all the files are combined after being built anyway, so it won't be any more space.
Doing things with files is likely to be quite common with this tool. I could foresee us having more utility functions in here, though I can't think of any specific examples right now. I think this gives us a good structure if we want to add more. I wouldn't consider that method belonging to any other modules

**Comment by MattLish** [comment]
I decided to remove the problem altogether and just use https://www.npmjs.com/package/junk instead and delete that file

**Comment by MattLish** [comment]
@RingComics I've updated to remove both of those files.

Also, this won't close #291 but once this is merged, 291 will be trivial because this contains an event to disable actions.

**Review by thomasblasquez — APPROVED** [review]
Certified dope af

**Comment by RingComics** [comment]
I will test it when I get home before approving and merging.

**Pull request merged**